buildSrc/src/main/kotlin/io/spine/internal/gradle/javascript/JsEnvironment.kt

```diff
@@ -0,0 +1,234 @@
+
+    /**
+     * An access token that allows installation and/or publishing modules.
+     *
+     * Default value is read from the environmental variable - `NPM_TOKEN`.
+     * "PUBLISHING_FORBIDDEN" stub value would be assigned in case `NPM_TOKEN` variable is not set.
+     *
+     * During installation a token is required only if dependencies
+     * from private repositories are used.
+     *
+     * See [Creating and viewing access tokens | npm Docs](https://docs.npmjs.com/creating-and-viewing-access-tokens)
+     */
+    val npmAuthToken: String
+        get() = System.getenv("NPM_TOKEN") ?: "PUBLISHING_FORBIDDEN"
+
+    /**
+     * A directory where `npm` puts downloaded module's dependencies.
+     *
+     * Default value: "projectDir/node_modules".
+     */
+    val nodeModules: File
+        get() = projectDir.resolve("node_modules")
+
+    /**
+     * Module's descriptor used by `npm`.
+     *
+     * Default value: "projectDir/package.json".
+     */
+    val packageJson: File
+        get() = projectDir.resolve("package.json")
+
+    /**
+     * `npm` gets its configuration settings from the command line, environment variables,
+     * and `npmrc` files.
+     *
+     * Default value: "projectDir/.npmrc".
+     *
+     * See [npmrc | npm Docs](https://docs.npmjs.com/cli/v8/configuring-npm/npmrc)
+     */
+    val npmrc: File
+        get() = projectDir.resolve(".npmrc")
+
+    /**
+     * A cache directory in which `nyc` tool outputs raw coverage report.
+     *
+     * See [istanbuljs/nyc](https://github.com/istanbuljs/nyc)
+     */
+    val nycOutput: File
+        get() = projectDir.resolve(".nyc_output")
+
+    /**
+     * A directory in which `webpack` would put a ready-to-use bundle.
+     *
+     * See [webpack - npm](https://www.npmjs.com/package/webpack)
+     */
+    val webPackOutput: File
+        get() = projectDir.resolve("dist")
+
+    /**
+     * A directory where bundled artifacts for further publishing would be prepared.
+     */
+    val webpackPublicationDir: File
+        get() = publicationDir.resolve("dist")
+
+
+    // Code generation.
+
+
+    /**
+     * Name of a directory that contains generated code.
+     */
+    val genProtoDirName: String
+        get() = "proto"
+
+    /**
+     * Directory with production Protobuf messages compiled into JavaScript.
+     */
+    val genProtoMain: File
+        get() = srcDir.resolve(genProtoDirName)
+
+    /**
+     * Directory with test Protobuf messages compiled into JavaScript.
+     */
+    val genProtoTest: File
+        get() = testSrcDir.resolve(genProtoDirName)
+}
+
+/**
+ * Allows overriding [JsEnvironment]'s defaults.
+ *
+ * All of declared properties can be split into two groups:
+ *
+ *  1. The ones that *define* something - can be overridden;
+ *  2. The ones that *describe* something - can not be overridden.
+ *
+ *  Overriding of "describing" properties leads to inconsistency with expectations. They are not
```

Comment:
Please rephrase this section. Something (which probably should tell how "describing" are used, and why they are NOT used by `npm`) is missing.

Also:
```
`npm` tool
```
and 

```
JS
```